### PR TITLE
Inject ReloadableConfig into JsonRPCServer

### PR DESCRIPTION
**Pull request by hmoniz**
This injects the ReloadableConfig as a constructor parameter into the JsonRPCServer as opposed to having the JsonRPCServer constructing it. The allows the ReloadableConfig to be resued by other components. In particular, it will allow the CheckpointSubsystem to use it.

### CODE REVIEW

**Comment by hmoniz** [comment]
Current dependencies on/for this PR:
* main
  * **PR #114** <a href="https://app.graphite.dev/github/pr/consensus-shipyard/ipc-agent/114" target="_blank">Graphite</a>  👈
    * **PR #115** <a href="https://app.graphite.dev/github/pr/consensus-shipyard/ipc-agent/115" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/consensus-shipyard/ipc-agent/114?utm_source=stack-comment).

File: src/server/handlers/config.rs [hunk 1]
```diff
@@ -36,9 +32,9 @@ impl JsonRPCRequestHandler for ReloadConfigHandler {
     async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
         log::info!("received request to reload config: {request:?}");
 
-        let path = request
-            .path
-            .unwrap_or_else(|| self.default_config_path.clone());
-        self.config.reload(path).await
+        if request.path.is_some() {
+            self.config.set_path(request.path.unwrap());
```
Comment (adlrocha) [inline src/server/handlers/config.rs:36]: Just to double-check, when we start the daemon if no config path is specified the default one is used to construct the `ReloadConfig` right?

**Review by adlrocha — APPROVED** [review]
LGTM, but I would like @cryptoAtwill to have a look before merging that he was the writing this piece of code. 

BTW, remember making the linter happy before merge :pray:

File: src/manager/checkpoint.rs [hunk 2]
```diff
@@ -33,16 +33,15 @@ use crate::lotus::LotusClient;
...
     /// The subsystem uses a `ReloadableConfig` to ensure that, at all, times, the subnets under
     /// management are those in the latest version of the config.
     config: ReloadableConfig,
 }
 
 impl CheckpointSubsystem {
     /// Creates a new `CheckpointSubsystem` with a configuration `config`.
-    #[allow(dead_code)]
-    fn new(config: ReloadableConfig) -> Self {
+    pub fn new(config: ReloadableConfig) -> Self {
```
Comment (cryptoAtwill) [inline src/manager/checkpoint.rs:44]: Here should have `Arc` for the `ReloadableConfig`?

**Review by cryptoAtwill — APPROVED** [review]

File: src/manager/checkpoint.rs (hunk 2 above)
Comment (hmoniz) [reply src/manager/checkpoint.rs:44]: Yes!

File: src/server/handlers/config.rs (hunk 1 above)
Comment (hmoniz) [reply src/server/handlers/config.rs:36]: It uses the one already present in `ReloadableConfig`.

**Pull request merged**